File: content/examples/other/leetcode/problems/number-of-1-bits/example-1/solution.js
function hammingWeightScan(n) {
  n = n >>> 0;
  let count = 0;
  for (let i = 0; i < 32; i += 1) {
    count += (n >>> i) & 1;
  }
  return count;
}

if (require.main === module) {
  console.log(hammingWeightScan(0b00000000000000000000000000001011));
}

module.exports = { hammingWeightScan };

File: content/examples/other/leetcode/problems/number-of-1-bits/example-1/solution.py
def hamming_weight_scan(n):
    n &= 0xFFFFFFFF
    count = 0
    for i in range(32):
        count += (n >> i) & 1
    return count


if __name__ == "__main__":
    print(hamming_weight_scan(0b00000000000000000000000000001011))
